### PR TITLE
initial proposed indexer api

### PR DESCRIPTION
**Pull request by sbezverk**
Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

### CODE REVIEW

File: indexer/indexer.proto [hunk 1]
```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+
+package indexer;
+option go_package=".;indexer";
+
+message FullIndex {
```
Comment (ahsalam) [inline indexer/indexer.proto]: Just curious if there was a reason to call messages as FullIndex and PartialIndex

**Review by giulio-sido — APPROVED** [review]

File: indexer/indexer.proto (hunk 1 above)
Comment (sbezverk) [reply indexer/indexer.proto]: In the latest revision It is `FullIndex` and `NodeIndex`. Full index request is sent to trigger assignment the interface ids to all nodes in the topology. Node index triggers assignment of ids for a single node/router. I am opened for alternative names.

Comment (ahsalam) [reply indexer/indexer.proto]: What do you think about "IDAlgoFull" and "IDAlgoNode" ? 
I'm fine with current names as well. Just wanted the request to have a more self description name

**Comment by sbezverk** [comment]
@ahsalam WDYT?

**Comment by ahsalam** [comment]
Looks good. Thanks!

**Pull request merged**